File: backend/src/main/java/com/example/demo/common/Result.java
package com.example.demo.common;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

/**
 * 统一响应包装类 —— 所有 REST 接口返回 {code, message, data[, errors]}
 * code: 200=成功 201=创建成功 400=业务/参数错误
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Result<T> {

    private int code;
    private String message;
    private T data;

    /** 参数校验失败时的字段错误信息（可选） */
    private Map<String, String> errors;

    public static <T> Result<T> success(T data) {
        return Result.<T>builder().code(200).message("success").data(data).build();
    }

    public static <T> Result<T> success(String message, T data) {
        return Result.<T>builder().code(200).message(message).data(data).build();
    }

    public static <T> Result<T> created(T data) {
        return Result.<T>builder().code(201).message("创建成功").data(data).build();
    }

    public static <T> Result<T> updated(T data) {
        return Result.<T>builder().code(200).message("更新成功").data(data).build();
    }

    public static <T> Result<T> deleted() {
        return Result.<T>builder().code(200).message("删除成功").build();
    }

    public static <T> Result<T> error(int code, String message) {
        return Result.<T>builder().code(code).message(message).build();
    }

    public static <T> Result<T> error(int code, String message, Map<String, String> errors) {
        return Result.<T>builder().code(code).message(message).errors(errors).build();
    }
}

File: backend/fastapi-app/app/core/result.py
"""统一响应包装 —— 对应 Java 的 Result<T>：所有接口返回 {code, message, data[, errors]}"""
from typing import Any, Generic, TypeVar

from pydantic import BaseModel

T = TypeVar("T")


class Result(BaseModel, Generic[T]):
    """code: 200=成功 201=创建成功 400=业务/参数错误 401=未授权"""

    code: int
    message: str
    data: T | None = None
    errors: dict[str, str] | None = None

    @classmethod
    def success(cls, *args, **kwargs) -> "Result":
        """兼容 success(data) 与 success(message, data) 两种调用（对应 Java 版两个重载）"""
        if kwargs:
            return cls(code=200, message=kwargs.get("message", "success"), data=kwargs.get("data"))
        if len(args) == 2:
            message, data = args
            return cls(code=200, message=message, data=data)
        if len(args) == 1:
            return cls(code=200, message="success", data=args[0])
        return cls(code=200, message="success", data=None)

    @classmethod
    def created(cls, data: Any = None) -> "Result":
        return cls(code=201, message="创建成功", data=data)

    @classmethod
    def updated(cls, data: Any = None) -> "Result":
        return cls(code=200, message="更新成功", data=data)

    @classmethod
    def deleted(cls) -> "Result":
        return cls(code=200, message="删除成功", data=None)

    @classmethod
    def error(cls, code: int, message: str, errors: dict[str, str] | None = None) -> "Result":
        return cls(code=code, message=message, data=None, errors=errors)
